src/evo/creditpool.cpp

```diff
@@ -202,10 +203,11 @@ CCreditPool CCreditPoolManager::GetCreditPool(const CBlockIndex* block_index, co
     std::stack<const CBlockIndex *> to_calculate;
 
     std::optional<CCreditPool> poolTmp;
-    while (!(poolTmp = GetFromCache(block_index)).has_value()) {
+    while (block_index != nullptr && !(poolTmp = GetFromCache(*block_index)).has_value()) {
         to_calculate.push(block_index);
         block_index = block_index->pprev;
     }
+    if (block_index == nullptr) poolTmp = CCreditPool{};
```

Comment:
`to_calculate` would be still processed. `poolTmp` is a first pool that calculation starts from. If by some reason we achieved the genesis block without any known credit pool - we start from empty credit pool.
That is necessary line because in next time `*poolTmp` is called and if this std::optional has nothing it maybe UB.

This condition doesn't change flow of function but just a guard for case if previous block doesn't exist.

Reply:
oh, right 🙈